pydantic/validators.py

```diff
@@ -472,6 +476,13 @@ def constr_strip_whitespace(v: 'StrBytes', field: 'ModelField', config: 'BaseCon
     return v
 
 
+def constr_lower(v: 'StrBytes', field: 'ModelField', config: 'BaseConfig') -> 'StrBytes':
+    lower = field.type_.lower_str or config.anystr_lower
+    if lower:
+        v = v.strip()
```

Comment:
```suggestion
        v = v.lower()
```

Reply:
You should probably add a test for `constr`

Reply:
thanks. I somehow missed that :+1: